test/functorch/test_control_flow.py

```diff
@@ -1174,6 +1203,225 @@ def fwbw(map_op, f, x, y):
         fake_outs = fwbw(_fake_map, f, x, y)
         self.assertEqual(true_outs, fake_outs)
 
+    @unittest.skipIf(not torch.cuda.is_available(), "Test requires CUDA.")
+    def test_generic_associative_scan_simple(self):
+        import random
+
+        def add(x: torch.Tensor, y: torch.Tensor):
+            return x + y
+
+        def mul(x: torch.Tensor, y: torch.Tensor):
+            return x * y
+
+        x = torch.randn(3, 10, 2, device=torch.device("cuda"))
+        for reverse in [False, True]:
+            for op, op_pt in [(add, torch.cumsum), (mul, torch.cumprod)]:
+                cumsum1 = associative_scan(op, x, 0, reverse=reverse)
+                cumsum_exp = _fake_associative_scan(op, x, 0, reverse=reverse)
+                self.assertEqual(cumsum1, cumsum_exp)
+                if not reverse:
+                    cumsum_exp_PT = op_pt(x, 0)
+                    self.assertEqual(cumsum1, cumsum_exp_PT)
+                else:
+                    # Check whether the graph contains at least two flip operations
+                    def f(op, inp, dim):
+                        result = associative_scan(op, inp, dim)
+                        return result
+
+                    # gm = make_fx(f, tracing_mode="symbolic")(op, x, 0)
```

Comment:
This is because associative_scan is calling torch.compile with default "inductor" backend, we don't expect make_fx this compiled fn gives us the graph.

Can you try torch.compile(f, backend=EagerAndRecordGraphs())? Can follow `test_cond_simple_with_linear_compile_check_graph`.

Reply:
Forget about this. This doesn't work because pointwise associative scan didn't implement eager backend.